web/client/utils/FileUtils.js

```diff
@@ -23,14 +32,55 @@ const FileUtils = {
     shpToGeoJSON: function(zipBuffer) {
         return [].concat(shp.parseZip(zipBuffer));
     },
+    kmlToGeoJSON: function(xml) {
+        let geoJSON = [].concat(tj.kml(xml));
```

Comment:
const geoJSON = [].concat(tj.kml(xml).map(item => assign({}, item, {fileName: xml.getElementsByTagName('name')[0].innerHTML})));

everywhere please, don't use forEach if you can use map / reduce